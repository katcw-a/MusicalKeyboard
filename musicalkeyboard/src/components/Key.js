import React from 'react';
import '../App.css';

import Sound from 'react-sound';
import testSound from '../samples/grand_piano/39148__jobro__piano-ff-001.wav'

class Key extends React.Component{
  constructor(props){
    super(props);
    this.state = {
      key: "",
      file: "",
      selected: false
    }
    this.keyPressed = this.keyPressed.bind(this);
    this.clearStroke = this.clearStroke.bind(this);
  }

  keyPressed (event){
    var activeKey = this.props.keys.filter(function(key){
      return key.id === event.keyCode
    });
    this.setState({
      key: activeKey,
      selected: true
    });
  }

  clearStroke (){
    this.setState({
      key: "",
      selected: false
    });
  }

  componentDidMount(){
    document.addEventListener("keydown", this.keyPressed, false);
    document.addEventListener("keyup", this.clearStroke, false);
  }

  render() {
    var firstRowKeys = this.props.keys.filter(function(key){
      return key.row === "first";
    })
    .map(function(key){
      return <div className="Key-inline" key={key.id}>{key.character}</div>
    });
    var secondRowKeys = this.props.keys.filter(function(key){
      return key.row === "second";
    })
    .map(function(key){
      return <div className="Key-inline" key={key.id}>{key.character}</div>
    });
    var thirdRowKeys = this.props.keys.filter(function(key){
      return key.row === "third";
    })
    .map(function(key){
      return <div className="Key-inline" key={key.id}>{key.character}</div>
    });
    return(
      <React.Fragment>
        <h1 className="Key">{firstRowKeys}</h1>
        <h1 className="Key">{secondRowKeys}</h1>
        <h1 className="Key">{thirdRowKeys}</h1>
        <Sound
          url={testSound}
          playStatus={Sound.status.PLAYING}
          onLoading={this.handleSongLoading}
          onPlaying={this.handleSongPlaying}
          onFinishedPlaying={this.handleSongFinishedPlaying}
        />
    </React.Fragment>
    );
  }
}

export default Key;