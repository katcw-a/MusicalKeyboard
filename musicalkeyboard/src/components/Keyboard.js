import React from 'react';
import Key from './Key';
import '../App.css';

class Keyboard extends React.Component{
  render(){
    var keys = [
    {id:81, row: "first", character: "q", file: "Empty"},
    {id:87, row: "first", character: "w", file: "Empty"},
    {id:69, row: "first", character: "e", file: "Empty"},
    {id:82, row: "first", character: "r", file: "Empty"},
    {id:84, row: "first", character: "t", file: "Empty"},
    {id:89, row: "first", character: "y", file: "Empty"},
    {id:85, row: "first", character: "u", file: "Empty"},
    {id:73, row: "first", character: "i", file: "Empty"},
    {id:79, row: "first", character: "o", file: "Empty"},
    {id:80, row: "first", character: "p", file: "Empty"},
    {id:65, row: "second", character: "a", file: "Empty"},
    {id:83, row: "second", character: "s", file: "Empty"},
    {id:68, row: "second", character: "d", file: "39148__jobro__piano-ff-001.wav"},
    {id:70, row: "second", character: "f", file: "Empty"},
    {id:71, row: "second", character: "g", file: "Empty"},
    {id:72, row: "second", character: "h", file: "Empty"},
    {id:74, row: "second", character: "j", file: "Empty"},
    {id:75, row: "second", character: "k", file: "Empty"},
    {id:76, row: "second", character: "l", file: "Empty"},
    {id:90, row: "third", character: "z", file: "Empty"},
    {id:88, row: "third", character: "x", file: "Empty"},
    {id:67, row: "third", character: "c", file: "Empty"},
    {id:86, row: "third", character: "v", file: "Empty"},
    {id:66, row: "third", character: "b", file: "Empty"},
    {id:78, row: "third", character: "n", file: "Empty"},
    {id:77, row: "third", character: "m", file: "Empty"},
    ];
    return(
      <div>
        <Key keys={keys}/>
      </div>
    )
  }
}

export default Keyboard;