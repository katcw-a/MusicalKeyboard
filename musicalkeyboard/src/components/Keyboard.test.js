import Keyboard from './Keyboard';
import React from 'react';
import { mount, shallow } from 'enzyme';
import Key from './Key';

var keys = [];
for (var i = 0; i < 6; i++){
  keys.push({
    id: i,
    row: "first", 
    character: "q", 
    file: "Empty"
  });
}

describe('Keyboard', () => {

  it('renders Key in Keyboard', () => {
    const wrapper = shallow(<Keyboard keys={keys}/>);
    expect(wrapper.find(Key)).toHaveLength(1);
  });
});
