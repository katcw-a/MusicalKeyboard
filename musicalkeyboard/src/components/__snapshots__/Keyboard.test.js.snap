// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keyboard should render the Keyboard components 1`] = `
<div>
  <Key
    keys={
      Array [
        Object {
          "character": "q",
          "file": "Empty",
          "id": 81,
          "row": "first",
        },
        Object {
          "character": "w",
          "file": "Empty",
          "id": 87,
          "row": "first",
        },
        Object {
          "character": "e",
          "file": "Empty",
          "id": 69,
          "row": "first",
        },
        Object {
          "character": "r",
          "file": "Empty",
          "id": 82,
          "row": "first",
        },
        Object {
          "character": "t",
          "file": "Empty",
          "id": 84,
          "row": "first",
        },
        Object {
          "character": "y",
          "file": "Empty",
          "id": 89,
          "row": "first",
        },
        Object {
          "character": "u",
          "file": "Empty",
          "id": 85,
          "row": "first",
        },
        Object {
          "character": "i",
          "file": "Empty",
          "id": 73,
          "row": "first",
        },
        Object {
          "character": "o",
          "file": "Empty",
          "id": 79,
          "row": "first",
        },
        Object {
          "character": "p",
          "file": "Empty",
          "id": 80,
          "row": "first",
        },
        Object {
          "character": "a",
          "file": "Empty",
          "id": 65,
          "row": "second",
        },
        Object {
          "character": "s",
          "file": "Empty",
          "id": 83,
          "row": "second",
        },
        Object {
          "character": "d",
          "file": "39148__jobro__piano-ff-001.wav",
          "id": 68,
          "row": "second",
        },
        Object {
          "character": "f",
          "file": "Empty",
          "id": 70,
          "row": "second",
        },
        Object {
          "character": "g",
          "file": "Empty",
          "id": 71,
          "row": "second",
        },
        Object {
          "character": "h",
          "file": "Empty",
          "id": 72,
          "row": "second",
        },
        Object {
          "character": "j",
          "file": "Empty",
          "id": 74,
          "row": "second",
        },
        Object {
          "character": "k",
          "file": "Empty",
          "id": 75,
          "row": "second",
        },
        Object {
          "character": "l",
          "file": "Empty",
          "id": 76,
          "row": "second",
        },
        Object {
          "character": "z",
          "file": "Empty",
          "id": 90,
          "row": "third",
        },
        Object {
          "character": "x",
          "file": "Empty",
          "id": 88,
          "row": "third",
        },
        Object {
          "character": "c",
          "file": "Empty",
          "id": 67,
          "row": "third",
        },
        Object {
          "character": "v",
          "file": "Empty",
          "id": 86,
          "row": "third",
        },
        Object {
          "character": "b",
          "file": "Empty",
          "id": 66,
          "row": "third",
        },
        Object {
          "character": "n",
          "file": "Empty",
          "id": 78,
          "row": "third",
        },
        Object {
          "character": "m",
          "file": "Empty",
          "id": 77,
          "row": "third",
        },
      ]
    }
  />
</div>
`;
